init.go

```diff
@@ -0,0 +1,28 @@
+package allure
+
+import (
+	"fmt"
+	"github.com/jtolds/gls"
+	"log"
+	"os"
+)
+
+var (
+	ctxMgr     *gls.ContextManager
+	wsd        string
+	resultPath string
+)
+
+const (
+	resultsPathEnvKey = "ALLURE_RESULTS_PATH"
+	nodeKey           = "current_step_container"
+)
+
+func init() {
+	ctxMgr = gls.NewContextManager()
+	wsd = os.Getenv(resultsPathEnvKey)
```

Comment:
Moving this in the init() method breaks the report generation because the variable environment can be initialized by the code of the project using allure-go, and this code can be executed after the inits.
In a global way, we should avoid inits as much as possible.
The initial code is also wrong, wsd stands for workspace directory, but ALLURE_RESULTS_PATH is not necessarily equal to workspace. We should use another env variable to define the workspace.
wsd should then be initialized in (r *result) setLabels() method (the lowest level where it is required).
resultPath should then be initialized in (r *result) writeResultsFile()

Reply:
A made the changes and tested it, it works. Merging the branch.